pkg/sif/create.go

```diff
@@ -676,3 +677,41 @@ func (f *FileImage) SetPrimPart(id uint32, opts ...SetOpt) error {
 
 	return nil
 }
+
+// SetMetadata writes the mutated descriptors and returns an error if any.
```

Comment:
This doesn't seem accurate? Probably should be something like:

```suggestion
// SetMetadata the metadata of the data object with id to md, according to opts.
//
// By default, the image/object modification times are set to the current time for
// non-deterministic images, and unset otherwise. To override this, consider using
// OptSetDeterministic or OptSetWithTime.
```